vars/artifactSetVersion.groovy

```diff
@@ -177,8 +177,36 @@ void call(Map parameters = [:], Closure body = null) {
                 error "[${STEP_NAME}]git commit and tag failed: ${e}"
             }
 
-            sshagent([config.gitSshKeyCredentialsId]) {
-                sh "git push ${config.gitSshUrl} ${config.tagPrefix}${newVersion}"
+            if(config.gitSshUrl) {
+                sshagent([config.gitSshKeyCredentialsId]) {
+                    sh "git push ${config.gitSshUrl} ${config.tagPrefix}${newVersion}"
+                }
+            } else if(config.gitHttpsUrl) {
+                 withCredentials([usernamePassword(
+                     credentialsId: config.gitCredentialsId,
+                     passwordVariable: 'PASSWORD',
+                     usernameVariable: 'USERNAME')]) {
+
+                     // Problem: when we encode the username it is not replaced by stars
+                     // in the log by surrounding withCredentials. For the user name that
+                     // might be just well enough. But we need the same for the password.
+                     // And for passwords that is a no-go.
```

Comment:
a `withEnv` block might help if you then later on just consume the environment variable in the git call

Reply:
yes, I will try that.

Reply:
@OliverNocon I tried to use the withEnv closure but it does not help. What I tried is basically

```
def gitUrlWithCredentials = "${prefix}\${XXUSER}:\${XXPASSWORD}@example.org/myGitRepo.git"
withEnv(["XXUSER=${USERNAME_ENCODED}", 
         "XXPASSWORD=${PASSWORD}"]) {
    sh script: "git push ${gitUrlWithCredentials} ${commonPipelineEnvironment.getArtifactVersion()}"
}
```

Expected result according to how I understood you comment above:

```
[Pipeline] sh
+ git push https://${XXUSER}:${XXPASSWORD}@example.org/myGitRepo.git 1.0.0-theAppendix
```

Actual result:

```
[Pipeline] sh
+ git push https://my%40User%40Name:myPassword@example.org/myGitRepo.git 1.0.0-theAppendix
```

Can you provide some more details about your suggestion? Is there some working example for that?

Reply:
@OliverNocon: my question above wrt hiding encoded secrets has been resolved.
Add `#!/bin/bash -e` (no `x`) at the beginning. The git push itself has a `--quiet` option making git push silent so that git itself does not tell us what is going on. As long as we do not receive a exception from the sh call we can assume the git push action succeeded.

Nevertheless: would be cool to get your approach also running in order to learn how it can be done otherwise.

Reply:
when using `withEnv{}` and then later using $VARIABLE (making sure that groovy does not replace it)  should do the job IMO ...

Reply:
I tried that, but it did not work. In the log issues by `sh` the variable was already expanded. And no: I did not make the mistake to mix-up the different levels of expansing: I used "\${VAR_NAME}" which ends up in getting the variable expanded on shell level, instead of "${VAR_NAME}" which expands on groovy level. I guess the current approach does the job, nevertheless it would be interesting to see your approach "in action".